File: frontend/src/api/product/dto/ProductDto.ts
import { NutrientDto } from "./NutrientDto";
import { CategorySimpleDto } from "./CategorySimpleDto";
import { AllergenDto } from "./AllergenDto";

export interface ProductDto {
  id: number;
  name: string;
  imageUrl: string;
  price: number;
  weight: number;
  nutrients: NutrientDto; // Nutritional information
  description?: string; // Optional
  isNew: boolean; // Indicates if the product is new
  supplierId: number;
  categories: CategorySimpleDto[]; // List of associated categories
  allergens: AllergenDto[]; // List of associated allergens
}

File: services/product-service/src/main/java/com/mealmap/productservice/core/dto/product/ProductDto.java
package com.mealmap.productservice.core.dto.product;

import com.mealmap.productservice.core.dto.allergen.AllergenDto;
import com.mealmap.productservice.core.dto.category.CategorySimpleDto;
import com.mealmap.productservice.core.dto.nutrient.NutrientDto;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Информация о продукте")
public class ProductDto {
    private Long id;

    private String name;

    private String imageUrl;

    private Integer price;

    private Integer weight;

    private NutrientDto nutrients;

    private String description;

    private Boolean isNew;

    private Integer supplierId;

    private List<CategorySimpleDto> categories;

    private List<AllergenDto> allergens;
}
